zcash_transparent/src/builder.rs

```diff
@@ -378,4 +451,365 @@ impl Bundle<Unauthorized> {
             authorization: Authorized,
         })
     }
+
+    /// Prepares the bundle for staged application of external signatures.
+    ///
+    /// This method computes the sighash for each transparent input using the provided
+    /// `calculate_sighash` closure.
+    #[cfg(feature = "transparent-inputs")]
+    pub fn prepare_transparent_signatures<F>(
+        self,
+        calculate_sighash: F,
+        secp_ctx: &secp256k1::Secp256k1<secp256k1::VerifyOnly>,
+    ) -> Result<TransparentSignatureContext<secp256k1::VerifyOnly>, Error>
+    where
+        F: Fn(SignableInput) -> [u8; 32], // The closure's signature
+    {
+        // Compute the sighashes for all inputs using the provided closure.
+        let sighashes = self
+            .authorization
+            .inputs
+            .iter()
+            .enumerate()
+            .map(|(index, info)| {
+                calculate_sighash(SignableInput {
+                    hash_type: SighashType::ALL,
+                    index,
+                    // for p2pkh, script_code is always the same as script_pubkey
+                    script_code: &info.coin.script_pubkey,
+                    script_pubkey: &info.coin.script_pubkey,
+                    value: info.coin.value,
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let num_inputs = self.authorization.inputs.len();
+        Ok(TransparentSignatureContext {
+            original_vin_unauthorized: self.vin,
+            original_vout: self.vout,
+            authorization_inputs: self.authorization.inputs,
+            sighashes, // The newly computed and owned Vec of sighashes
+            secp_ctx,
+            final_script_sigs: vec![None; num_inputs],
+        })
+    }
+}
+
+#[cfg(feature = "transparent-inputs")]
+impl<'a> TransparentSignatureContext<'a, secp256k1::VerifyOnly> {
+    /// Appends a new batch of external signatures to the transparent inputs.
+    ///
+    /// This method iterates through the provided signatures, applying each one to the
+    /// single input for which it is valid.
+    pub fn append_external_signatures(
+        self, // Takes ownership of self
```

Comment:
```suggestion
        self,
```
This is also obvious by being spelled just `self`. Possibly needs a rustfmt after applying the change.